elki/src/main/java/ch/ethz/globis/pht/util/Refs.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2011-2015 ETH Zurich. All Rights Reserved.
+ *
+ * This software is the proprietary information of ETH Zurich.
+ * Use is subject to license terms.
+ */
+package ch.ethz.globis.pht.util;
+
+import java.util.Arrays;
+
+import ch.ethz.globis.pht.PhTreeHelper;
+
+
+public class Refs {
+	
+	private static final int BYTES_PER_UNIT = 4;//bytes
+//	private static int statAExpand;
+//	private static int statACreate;
+//	private static int statATrim;
+	
+	public static final Object[] EMPTY_REF_ARRAY = {};
+    private static final ArrayPool POOL = 
+    		new ArrayPool(PhTreeHelper.ARRAY_POOLING_MAX_ARRAY_SIZE, 
+    				PhTreeHelper.ARRAY_POOLING_POOL_SIZE);
+
+    
+    private static class ArrayPool {
```

Comment:
Well:

1) Pooling can be disabled by setting `PhTreeHelper.ARRAY_POOLING_POOL_SIZE = 0`
2) Pooling wasn't used (or even implemented) in any paper I published so far, so no research is compromised.
3) It improves performance for the PhTree during insert/delete by 10-20%. This is because almost every insert/delete comes an array resize. This is necessary to reduce memory consumption of the many small nodes (most nodes have only two entries). Hence I doubt that other trees would gain as much as the PhTree from using pooling, i.e. the difference may not be measurable.

Anyway, I don't mind at all if we put this class in a more public place so it being used by other trees as well. 

As a side note, the tree I provide here has been significantly improved in many areas over the version that was published in my SIGMOD paper. None of the improvements have been published yet, I'm not sure I will find time anytime soon to publish them (they are not really worth an own publication anyway).